extensions/amp-cxense-player/0.1/amp-cxense-player.js

```diff
@@ -0,0 +1,244 @@
+        /** @private @const {number} */
+      this.height_ = getLengthNumeral(this.element.getAttribute('height'));
+
+      if (!this.getPlaceholder()) {
+        this.buildImagePoster_();
+        this.loading_(true);
+      }
+    }
+
+    /** @override */
+    layoutCallback() {
+      const self = this;
+
+        /** @private @const {Element} */
+      this.iframe_ = this.element.ownerDocument.createElement('iframe');
+
+      this.iframe_.setAttribute('frameborder', '0');
+      this.iframe_.setAttribute('allowfullscreen', 'true');
+
+      this.iframe_.width = this.width_;
+      this.iframe_.height = this.height_;
+
+      this.iframe_.src = this.getIframeSrc_();
+      setStyles(this.iframe_, {display: 'none'});
+      this.element.appendChild(this.iframe_);
+
+        /** @private {boolean} */
+      this.mpfReady_ = false;
+
+        /** @private @const {!Promise} */
+      this.playerReadyPromise_ = new Promise(resolve => {
+            /** @private @const {function()} */
+        this.playerReadyResolver_ = iframe => {
+          self.mpfReady = true;
+          resolve(iframe);
+        };
+      });
+
+      this.handleFrameMessages_ = this.handleFrameMessages_.bind(this);
+      this.getWin().addEventListener('message', this.handleFrameMessages_);
+
+      return loadPromise(this.iframe_)
+            .then(() => this.playerReadyPromise_)
+            .then(ret => {
+              self.loading_(false);
+              setStyles(self.iframe_, {display: ''});
+              self.applyFillContent(self.iframe_);
+              return ret;
+            });
+    }
+
+    /** @override */
+    pauseCallback() {
+      this.postMessage_({type: 'mpf.video.pause'});
+      return true;
+    }
+
+    /** @override */
+    unlayoutCallback() {
+      this.getWin().removeEventListener('message', this.handleFrameMessages_);
+      this.iframe_.setAttribute('src', 'about:blank');
+      this.iframe_.parentNode.removeChild(this.iframe_);
+      return true;
+    }
+
+    /** @private */
+    handleFrameMessages_(event) {
+      if (event.origin != this.cxDefaults_.embedHost
+            || event.source != this.iframe_.contentWindow) {
+        return;
+      }
+      let data;
+      if (!event.data || event.data.indexOf('{') != 0) {
+        return;  // Doesn't look like JSON.
+      }
+      try {
+        data = JSON.parse(event.data);
+      } catch (unused) {
+        return; // We only process valid JSON.
+      }
+      // for now, we only need this event
+      if (data.type == 'mpf.ready') {
+        this.playerReadyResolver_(this.iframe_);
+      }
+    }
+
+    /** @private */
+    getIframeSrc_() {
+      const attrs = this.getDataAttributes_();
+      return addParamsToUrl(this.cxDefaults_.embedHost
+            + this.cxDefaults_.embedApp
+            , attrs);
+    }
+
+    /** @private */
+    getDataAttributes_() {
+      return extend(
+            this.cxDefaults_.attrs,
```

Comment:
> This mutates the default attributes,
Not really, remember I made a copy here

https://github.com/ampproject/amphtml/pull/3951#discussion_r70246001

also, `share.enable` can be overridden depending on the use case and the conditions of the videos.